### PR TITLE
feat: [PIE-4480]: refactor azure web app infra 

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

Refactor Azure web app infra and add target slot step

#### Screenshots

https://user-images.githubusercontent.com/106532291/179508411-e5146cf7-31a6-499e-8aef-629991993814.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [Y] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx [hunk 1]
```diff
@@ -97,6 +109,57 @@ export const AzureSlotDeploymentRef = (
                 />
               )}
             </div>
+            <div className={stepCss.divider} />
+            <div className={cx(stepCss.formGroup, stepCss.lg)}>
+              <FormInput.MultiTextInput
+                name="spec.webApp"
+                placeholder={'Specify web app name'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx:116]: Move this to strings file

File: src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx [hunk 2]
```diff
@@ -97,6 +109,57 @@ export const AzureSlotDeploymentRef = (
...
+                label={'Web App Name'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx:117]: Move this to strings file.

File: src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx [hunk 3]
```diff
@@ -97,6 +109,57 @@ export const AzureSlotDeploymentRef = (
...
+                    }
+                  }
+                  isReadonly={readonly}
+                />
+              )}
+            </div>
+            <div className={cx(stepCss.formGroup, stepCss.lg)}>
+              <FormInput.MultiTextInput
+                name="spec.deploymentSlot"
+                placeholder={'Specify deployment slot'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx:141]: Move this to strings file

File: src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx [hunk 4]
```diff
@@ -97,6 +109,57 @@ export const AzureSlotDeploymentRef = (
...
+                label={'Deployment Slot'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx:142]: Move this to strings file

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppSwapSlot/AzureWebAppSwapSlotInputStep.tsx [hunk 5]
```diff
@@ -45,6 +45,20 @@ export function AzureWebAppSwapSlotInputStepRef<T extends AzureWebAppSwapSlotDat
           </div>
         )
       }
+      {isRuntime(inputSetData?.template?.spec?.targetSlot as string) && (
+        <div className={cx(stepCss.formGroup, stepCss.md)}>
+          <FormInput.MultiTextInput
+            label={'Target Slot'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureWebAppSwapSlot/AzureWebAppSwapSlotInputStep.tsx:51]: Move this to strings file

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppSwapSlot/AzureWebAppSwapSlotRef.tsx [hunk 6]
```diff
@@ -76,6 +85,32 @@ export const AzureWebAppSwapSlotRef = (
                 disabled={readonly}
               />
             </div>
+            <div className={stepCss.divider} />
+            <div className={cx(stepCss.formGroup, stepCss.lg)}>
+              <FormInput.MultiTextInput
+                name="spec.targetSlot"
+                placeholder={'Specify target slot'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureWebAppSwapSlot/AzureWebAppSwapSlotRef.tsx:92]: Move this to strings file

File: src/modules/75-cd/components/PipelineSteps/AzureWebAppSwapSlot/AzureWebAppSwapSlotRef.tsx [hunk 7]
```diff
@@ -76,6 +85,32 @@ export const AzureWebAppSwapSlotRef = (
...
+                label={'Target Slot'}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AzureWebAppSwapSlot/AzureWebAppSwapSlotRef.tsx:93]: Move this to strings file

**Review by chetanvekariya-harness — APPROVED** [review]
Move hard-coded text to strings files in this PR or in the next PR.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx (hunk 1 above)
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentRef.tsx:116]: same for placeholder

File: src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentVariableView.tsx [hunk 8]
```diff
@@ -6,22 +6,25 @@
...
   metadataMap,
   initialValues
 }: AzureSlotDeploymentVariableStepProps): React.ReactElement {
+  const data: Record<any, any> = pick(get(variablesData, 'spec', {}), ['webApp', 'deploymentSlot'])
   return (
     <VariablesListTable
-      data={variablesData}
-      originalData={initialValues}
+      data={data}
+      originalData={initialValues as Record<string, any>}
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineSteps/AzureSlotDeployment/AzureSlotDeploymentVariableView.tsx:27]: why any?

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**